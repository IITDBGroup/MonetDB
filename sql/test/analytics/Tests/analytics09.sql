create table analytics (aa int, bb int);
insert into analytics values (15, 3), (3, 1), (2, 1), (5, 3), (NULL, 2), (3, 2), (4, 1), (6, 3), (8, 2), (NULL, 4);

select cast(sum(1) over () as bigint), rank() over (), nth_value(1, 1) over ();

select avg(sum(aa) over ()) from analytics;

select cast(sum(1) * count(*) over () as bigint);

select cast(sum(aa) * count(*) over () as bigint) from analytics;

select cast(aa * count(1) over () as bigint) from analytics;

select cast(sum(aa) * count(1) over () as bigint) from analytics;

select cast(sum(aa) * count(1 + aa) / avg(1) over () as bigint) from analytics;

select avg(sum(aa)) over () from analytics;

select sum(cast(aa as double)) over (rows unbounded preceding) from analytics;

select sum(cast(aa as double)) over (range unbounded preceding) from analytics;

select avg(avg(aa)) over (rows unbounded preceding) from analytics;

select avg(avg(aa)) over (range unbounded preceding) from analytics;

select avg(sum(aa)) over (rows unbounded preceding) from analytics;

select avg(sum(aa)) over (range unbounded preceding) from analytics;

select avg(sum(aa)) over (), avg(avg(aa)) over () from analytics; --TODO support multiple aggregations within the same projection

select avg(sum(aa)) over (),
       cast(sum(aa) * count(case when bb < 2 then bb - 1 else bb + 1 end) / avg(1) over (rows between current row and current row) as bigint),
       avg(sum(aa)) over (rows unbounded preceding),
       avg(sum(aa)) over (range unbounded preceding) from analytics; --TODO support multiple aggregations within the same projection

select avg(sum(aa)) over () from analytics group by aa;

select avg(sum(aa)) over (),
       avg(sum(aa)) over (rows unbounded preceding),
       cast(sum(aa) * count(aa) / avg(aa) over (rows between current row and unbounded following) as bigint),
       avg(sum(aa)) over (range unbounded preceding) from analytics group by aa; --TODO support multiple aggregations within the same projection

select avg(sum(aa)) over (partition by bb) from analytics group by bb;

select cast(sum(aa) * 100 / sum(sum(aa)) over () as bigint) from analytics;

select cast(sum(aa) * 100 / sum(sum(aa)) over (partition by bb) as bigint) from analytics group by bb;

select cast(sum(aa) * 100 / sum(sum(aa)) over (partition by bb) as bigint) from analytics; --error, nesting aggregation functions

select cast(prod(sum(aa)) * count(1 + aa) / avg(null) over () as bigint) from analytics; --error, nesting aggregation functions

select rank() over (partition by case when aa > 5 then aa else aa + 5 end) from analytics; --TODO we don't support expressions in partition by as well group by statements

select rank() over (partition by sum(aa)) from analytics; --TODO we don't support expressions in partition by as well group by

drop table analytics;
