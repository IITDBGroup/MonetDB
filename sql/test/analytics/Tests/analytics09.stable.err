stderr of test 'analytics09` in directory 'sql/test/analytics` itself:


# 13:13:17 >  
# 13:13:17 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=37694" "--set" "mapi_usock=/var/tmp/mtest-1389/.s.monetdb.37694" "--set" "monet_prompt=" "--forcemito" "--dbpath=/home/ferreira/repositories/MonetDB-analytics/BUILD/var/MonetDB/mTests_sql_test_analytics" "--set" "embedded_c=true"
# 13:13:17 >  

# builtin opt 	gdk_dbpath = /home/ferreira/repositories/MonetDB-analytics/BUILD/var/monetdb5/dbfarm/demo
# builtin opt 	gdk_debug = 0
# builtin opt 	gdk_vmtrim = no
# builtin opt 	monet_prompt = >
# builtin opt 	monet_daemon = no
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	mapi_open = true
# cmdline opt 	mapi_port = 37694
# cmdline opt 	mapi_usock = /var/tmp/mtest-1389/.s.monetdb.37694
# cmdline opt 	monet_prompt = 
# cmdline opt 	gdk_dbpath = /home/ferreira/repositories/MonetDB-analytics/BUILD/var/MonetDB/mTests_sql_test_analytics
# cmdline opt 	embedded_c = true
# cmdline opt 	gdk_debug = 553648138

# 13:13:17 >  
# 13:13:17 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-1389" "--port=37694"
# 13:13:17 >  

MAPI  = (monetdb) /var/tmp/mtest-1389/.s.monetdb.37694
QUERY = select avg(sum(aa)) over (), avg(avg(aa)) over () from analytics; --TODO support multiple aggregations within the same projection
ERROR = !SELECT: cannot use non GROUP BY column 'aa' in query results without an aggregate function
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-1389/.s.monetdb.37694
QUERY = select avg(sum(aa)) over (),
               cast(sum(aa) * count(case when bb < 2 then bb - 1 else bb + 1 end) / avg(1) over (rows between current row and current row) as bigint),
               avg(sum(aa)) over (rows unbounded preceding),
               avg(sum(aa)) over (range unbounded preceding) from analytics; --TODO support multiple aggregations within the same projection
ERROR = !SELECT: cannot use non GROUP BY column 'aa' in query results without an aggregate function
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-1389/.s.monetdb.37694
QUERY = select avg(sum(aa)) over (),
               avg(sum(aa)) over (rows unbounded preceding),
               cast(sum(aa) * count(aa) / avg(aa) over (rows between current row and unbounded following) as bigint),
               avg(sum(aa)) over (range unbounded preceding) from analytics group by aa; --TODO support multiple aggregations within the same projection
ERROR = !Cannot use non GROUP BY column 'L4' in query results without an aggregate function
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-1389/.s.monetdb.37694
QUERY = select cast(sum(aa) * 100 / sum(sum(aa)) over (partition by bb) as bigint) from analytics; --error, nesting aggregation functions
ERROR = !SELECT: cannot use non GROUP BY column 'bb' in query results without an aggregate function
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-1389/.s.monetdb.37694
QUERY = select cast(prod(sum(aa)) * count(1 + aa) / avg(null) over () as bigint) from analytics; --error, nesting aggregation functions
ERROR = !SELECT: subquery result missing
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-1389/.s.monetdb.37694
QUERY = select rank() over (partition by case when aa > 5 then aa else aa + 5 end) from analytics; --TODO we don't support expressions in partition by as well group by statements
ERROR = !syntax error, unexpected CASE in: "select rank() over (partition by case"
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-1389/.s.monetdb.37694
QUERY = select rank() over (partition by sum(aa)) from analytics; --TODO we don't support expressions in partition by as well group by
ERROR = !syntax error, unexpected '(', expecting ')' in: "select rank() over (partition by sum("
CODE  = 42000

# 13:13:17 >  
# 13:13:17 >  "Done."
# 13:13:17 >  

