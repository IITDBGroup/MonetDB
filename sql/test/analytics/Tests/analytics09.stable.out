stdout of test 'analytics09` in directory 'sql/test/analytics` itself:


# 13:13:17 >  
# 13:13:17 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=37694" "--set" "mapi_usock=/var/tmp/mtest-1389/.s.monetdb.37694" "--set" "monet_prompt=" "--forcemito" "--dbpath=/home/ferreira/repositories/MonetDB-analytics/BUILD/var/MonetDB/mTests_sql_test_analytics" "--set" "embedded_c=true"
# 13:13:17 >  

# MonetDB 5 server v11.32.0
# This is an unreleased version
# Serving database 'mTests_sql_test_analytics', using 8 threads
# Compiled for x86_64-pc-linux-gnu/64bit
# Found 15.492 GiB available main-memory.
# Copyright (c) 1993 - July 2008 CWI.
# Copyright (c) August 2008 - 2018 MonetDB B.V., all rights reserved
# Visit https://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://wired-142.cwi.nl:37694/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-1389/.s.monetdb.37694
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded

Ready.

# 13:13:17 >  
# 13:13:17 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-1389" "--port=37694"
# 13:13:17 >  

#create table analytics (aa int, bb int);
#insert into analytics values (15, 3), (3, 1), (2, 1), (5, 3), (NULL, 2), (3, 2), (4, 1), (6, 3), (8, 2), (NULL, 4);
[ 10	]
#select cast(sum(1) over () as bigint), rank() over (), nth_value(1, 1) over ();
% .L6,	.L11,	.L15 # table_name
% L6,	L11,	L15 # name
% bigint,	int,	tinyint # type
% 1,	1,	1 # length
[ 1,	1,	1	]
#select avg(sum(aa) over ()) from analytics;
% sys.L10 # table_name
% L10 # name
% double # type
% 24 # length
[ 46	]
#select cast(sum(1) * count(*) over () as bigint);
% .L7 # table_name
% L7 # name
% bigint # type
% 1 # length
[ 1	]
#select cast(sum(aa) * count(*) over () as bigint) from analytics;
% sys.L7 # table_name
% L7 # name
% bigint # type
% 2 # length
[ 46	]
#select cast(aa * count(1) over () as bigint) from analytics;
% sys.L6 # table_name
% L6 # name
% bigint # type
% 3 # length
[ 150	]
[ 30	]
[ 20	]
[ 50	]
[ NULL	]
[ 30	]
[ 40	]
[ 60	]
[ 80	]
[ NULL	]
#select cast(sum(aa) * count(1) over () as bigint) from analytics;
% sys.L7 # table_name
% L7 # name
% bigint # type
% 2 # length
[ 46	]
#select cast(sum(aa) * count(1 + aa) / avg(1) over () as bigint) from analytics;
% sys.L10 # table_name
% L10 # name
% bigint # type
% 3 # length
[ 368	]
#select avg(sum(aa)) over () from analytics;
% sys.L6 # table_name
% L6 # name
% double # type
% 24 # length
[ 46	]
#select sum(cast(aa as double)) over (rows unbounded preceding) from analytics;
% sys.L5 # table_name
% L5 # name
% double # type
% 24 # length
[ 15	]
[ 18	]
[ 20	]
[ 25	]
[ 25	]
[ 28	]
[ 32	]
[ 38	]
[ 46	]
[ 46	]
#select sum(cast(aa as double)) over (range unbounded preceding) from analytics;
% sys.L5 # table_name
% L5 # name
% double # type
% 24 # length
[ 46	]
[ 46	]
[ 46	]
[ 46	]
[ 46	]
[ 46	]
[ 46	]
[ 46	]
[ 46	]
[ 46	]
#select avg(avg(aa)) over (rows unbounded preceding) from analytics;
% sys.L6 # table_name
% L6 # name
% double # type
% 24 # length
[ 5.75	]
#select avg(avg(aa)) over (range unbounded preceding) from analytics;
% sys.L6 # table_name
% L6 # name
% double # type
% 24 # length
[ 5.75	]
#select avg(sum(aa)) over (rows unbounded preceding) from analytics;
% sys.L6 # table_name
% L6 # name
% double # type
% 24 # length
[ 46	]
#select avg(sum(aa)) over (range unbounded preceding) from analytics;
% sys.L6 # table_name
% L6 # name
% double # type
% 24 # length
[ 46	]
#select avg(sum(aa)) over () from analytics group by aa;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 6.571428571	]
[ 6.571428571	]
[ 6.571428571	]
[ 6.571428571	]
[ 6.571428571	]
[ 6.571428571	]
[ 6.571428571	]
[ 6.571428571	]
#select avg(sum(aa)) over (partition by bb) from analytics group by bb;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 9	]
[ 11	]
[ 26	]
[ NULL	]
#select cast(sum(aa) * 100 / sum(sum(aa)) over () as bigint) from analytics;
% sys.L10 # table_name
% L10 # name
% bigint # type
% 3 # length
[ 100	]
#select cast(sum(aa) * 100 / sum(sum(aa)) over (partition by bb) as bigint) from analytics group by bb;
% sys.L7 # table_name
% L7 # name
% bigint # type
% 3 # length
[ 100	]
[ 100	]
[ 100	]
[ NULL	]
#drop table analytics;

# 13:13:17 >  
# 13:13:17 >  "Done."
# 13:13:17 >  

